http-ballerina/Module.md

```diff
@@ -0,0 +1,82 @@
+## Overview
+
+This module provides an implementation for connecting and interacting with HTTP and HTTP2 endpoints. It
+facilitates two types of network entry points as `Client` and `Listener`.
+
+### Client
+
+The `Client` is used to connect to and interact with HTTP endpoints. They support connection pooling and can be 
+configured to have a maximum number of active connections that can be made with the remote endpoint. The `Client` 
+activates connection eviction after a given idle period and also supports follow-redirects so that the users do not 
+have to manually handle 3xx HTTP status codes.
+
+The `Client` handles resilience in multiple ways such as load balancing, circuit breaking, endpoint timeouts, and a 
+retry mechanism.
+
+Load balancing is used in the round-robin or failover manner.
+
+When a failure occurs in the remote service, the client connections might wait for some time before a timeout occurs. 
+Awaiting requests consume resources in the system. Circuit Breakers are used to trip after a certain number of failed 
+requests to the remote service. Once a circuit breaker trips, it does not allow the client to send requests to the 
+remote service for a period of time.
+
+The Ballerina circuit breaker supports tripping on HTTP error status codes and I/O errors. Failure thresholds can be 
+configured based on a sliding window (e.g., 5 failures within 10 seconds). `Client` also supports a retry 
+mechanism that allows a client to resend failed requests periodically for a given number of times.
+
+The `Client` supports Server Name Indication (SNI), Certificate Revocation List (CRL), Online Certificate Status 
```

Comment:
Then we need a section for security and explain the rest as well in addition to SSL level stuff.

Reply:
Also I think talk bit about Caching